sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -97,6 +97,7 @@ func main() {
 	flag.String("virtualenv", "", "[obsolete] Virtual environment path to use")
 	flag.String("root", "", "[obsolete] Project root path to use")
 	flag.String("typechecker", "", "[obsolete] Use a typechecker to type check")
+	flag.String("toolchain", "pip", "[obsolete] Select the package manager to use for dependency management.")
```

Comment:
Curious, why are we adding a new option that's obsolete here?

Reply:
I had previously wondered about that too, Fraser explained here https://github.com/pulumi/pulumi/pull/15725#discussion_r1540785931

> Passing runtime flags as command line parameters to the language hosts is a deprecated form of passing them. You'll note that main doesn't do anything with these flags.
BUT the engine still sends these flags to language hosts for the time being as a back-compat workaround till all the hosts are updated, so the parser needs to recognize them so it doesn't fail on startup with "unrecognized argument".